scripts/generate_driver_wrappers.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+"""This script is required for the auto generation of the
+   psa_crypto_driver_wrappers.c file"""
+
+import sys
+import os
+import jinja2
+
+def render(template_path: str) -> str:
+    environment = jinja2.Environment(
+        loader=jinja2.FileSystemLoader(os.path.dirname(template_path)),
+        keep_trailing_newline=True)
+    template = environment.get_template(os.path.basename(template_path))
+    return template.render()
+
+N = len(sys.argv)
+if N != 2:
+# This is the Root directory.
+    ROOT_DIR = ""
+else:
+# Set the root based on the argument passed.
+    ROOT_DIR = sys.argv[1]
```

Comment:
Passing the root directory as an argument doesn't make much sense to me. But ok for now. Later, the various file names should be passed as options (`import argparse`).

Reply:
The idea of passing the root directory as an argument was to ensure we don't need to pass the arguments for file to be generated and template path for scripts from different places.
This is if  we know what  and where the template file is. So we just ensure we always run from the appropriate path with the ROOT_DIR being passed.

However, if we want to provide options for different template files to be used from custom paths, then yes, we will need to pass the arguments. 
Passing multiple files to the scripts in the way it is currently setup is honestly an eyesore .

Reply:
The most common usage for the generator will be to pass a list of driver descriptions, and otherwise keep default settings. So I think these should be the positional arguments.
```
scripts/generate_driver_wrappers.py /path/to/my/accelerator.json /path/to/my/secure-element.json /path/to/third-party/secure-element.json
```

Passing alternative templates, output paths, etc., are secondary settings mostly useful for unusual build systems. I only see one use case for changing both the template path and the output path at once, and that's if you run the script from outside the source tree, e.g. to run the script from a different version of Mbed TLS, which is definitely not a common use case. So I don't see the point of `ROOT_DIR`.

Reply:
I think we can rest this. Am rewriting this with argParse

Reply:
> The most common usage for the generator will be to pass a list of driver descriptions, and otherwise keep default settings. So I think these should be the positional arguments.
> 

While driver descriptions are for the next version, am planning on listing the order of the drivers in a json, and not pass it as an argument. Based on the order a concatenated json would be created, parsed and templated in that order.

Reply:
Having to create a file for the drivers doesn't sound convenient.

Reply:
(Again, this is unrelated to this PR) 
But the way the build scripts are setup, one will need to pass it in multiple places.
Isn't it more convenient to have a driver list file which can be edited by the user  to specify the order of the driver jsons.
eg.,
]#cat driverlist.json
["mbedtls_test_opaque_driver.json","mbedtls_test_transparent_driver.json"] 

The user does not have to create this file, it will be pre created, but the user can edit it with their preferred order of drivers. Sounds good no ?

Reply:
Not a comment for this patch, but another trick you can could possibly do is add an indexing \__init\__.py file which parses the files, and creates a dictionary when invoked, and the caller script imports everything as a module.

\__init\__.py
~~~
def parse_directory():
     somelogic()

if __name__ == '__main__':
    driver_map = parse_directory()
~~~

somescript.py
~~~
from templates_dir import driver_map

for map_name, map_data in driver_map.items():
    ....

~~~

That way the scipt can use any level of iterators to list valid configs and even bind it with argparse outputs, without having to hardcode any logic or file list. In addition the benefit is that init file knows that it will always be in the directory of the templates.